packages/node_modules/@ciscospark/plugin-memberships/src/memberships.js

```diff
@@ -24,6 +38,148 @@ import {SparkPlugin, Page} from '@ciscospark/spark-core';
+      return;
+    }
+
+    /* eslint-disable no-case-declarations */
+    // eslint-disable-next-line padding-line-between-statements
+    switch (activity.verb) {
+      case API_ACTIVITY_VERB.ADD:
+        const membershipCreatedEventData = this.getCreatedEvent(activity);
+        debug(`membership "created" payload: ${JSON.stringify(membershipCreatedEventData)}`);
+        this.trigger(MEMBERSHIP_CREATED, membershipCreatedEventData);
+        break;
+
+      case API_ACTIVITY_VERB.LEAVE:
+        const membershipDeletedEventData = this.getDeletedEvent(activity);
+        debug(`membership "deleted" payload: ${JSON.stringify(membershipDeletedEventData)}`);
+        this.trigger(MEMBERSHIP_DELETED, membershipDeletedEventData);
+        break;
+
+      case API_ACTIVITY_VERB.ACKNOWLEDGE:
+      case API_ACTIVITY_VERB.ADD_MODERATOR:
+      case API_ACTIVITY_VERB.REMOVE_MODERATOR:
+        const membershipUpdatedEventData = this.getUpdatedEvent(activity);
+        debug(`membership "updated" payload: ${JSON.stringify(membershipUpdatedEventData)}`);
+        this.trigger(MEMBERSHIP_UPDATED, membershipUpdatedEventData);
+        break;
+
+      default:
+        break;
+    }
+  },
+
+  /**
+   * Constructs event data object for the "membership created" event,
+   * adhering to Hydra's Webehook data for an "created" event on the "membership" resource.
+   * @param {Object} activity from mercury
+   * @returns {Object} constructed event
+   */
+  getCreatedEvent(activity) {
+    const sdkEvent = this.getCommonSdkEvent(activity);
+    sdkEvent.event = MEMBERSHIP_CREATED;
+    return sdkEvent;
+  },
+
+  /**
+   * Constructs event data object for the "membership deleted" event,
+   * adhering to Hydra's Webehook data for an "deleted" event on the "membership" resource.
+   * @param {Object} activity from mercury
+   * @returns {Object} constructed event
+   */
+  getDeletedEvent(activity) {
+    const sdkEvent = this.getCommonSdkEvent(activity);
+    sdkEvent.event = MEMBERSHIP_DELETED;
+    return sdkEvent;
+  },
+
+  /**
+   * Constructs event data object for the "membership updated" event,
+   * adhering to Hydra's Webehook data for an "updated" event on the "membership" resource.
+   * we add a field "lastSeenId" not yet available in the webhooks if this is an acknowledge actvtity
+   * @param {Object} activity from mercury
+   * @returns {Object} constructed event
+   */
+  getUpdatedEvent(activity) {
+    const sdkEvent = this.getCommonSdkEvent(activity);
+    sdkEvent.event = MEMBERSHIP_UPDATED;
+    if (activity.verb === API_ACTIVITY_VERB.ACKNOWLEDGE) {
+      sdkEvent.data.lastSeenId = constructHydraId(hydraTypes.MESSAGE, activity.object.id);
+    }
+    sdkEvent.data.created = activity.published;
+    return sdkEvent;
+  },
+
+  /**
+   * Constructs common event data object for all membership events
+   * @param {Object} activity from mercury
+   * @returns {Object} constructed event
+   */
+  getCommonSdkEvent(activity) {
```

Comment:
Is this envelope needed for the `messages` plugin too?

Reply:
It is.  Will discuss with you how to implement it there.